### PR TITLE
Added build script, build info and dockerfile for antlr4

### PR DESCRIPTION
**Pull request by muskaan2199**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

File: a/antlr4/Dockerfiles/ubi-8.5/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+FROM registry.access.redhat.com/ubi8 
+
+WORKDIR /opt/antlr4
+
+ARG ANTLR_VERSION="4.11.1"
```
Comment (kandarpamalipeddi) [inline a/antlr4/Dockerfiles/ubi-8.5/Dockerfile:5]: By defualt we send the `$PACKAGE_VERSION` argument to our build script and Docker file.
So in this case, we never send ANTLR_VERSION, hence it always build 4.11.1 version.

To fix this, either assign ANTLR_VERSION with `$PACKAGE_VERSION` in `build_info.json` file in `args` section or rename ANTLR_VERSION arg to PACKAGE_VERSION.

File: a/antlr4/Dockerfiles/ubi-8.5/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    && mvn -DskipTests install --projects tool --also-make \
+    && mv tool/target/antlr4-*-complete.jar antlr4-tool.jar \
+	&& cp antlr4-tool.jar /usr/local/lib/antlr4-tool.jar
+
+ARG user=appuser
+ARG group=appuser
+ARG uid=1000
+ARG gid=1000
+
+RUN adduser \
```
Comment (kandarpamalipeddi) [inline a/antlr4/Dockerfiles/ubi-8.5/Dockerfile:34]: We are creating a user but not using/setting it.
Can you please check it once?

File: a/antlr4/antlr4_v4.11.1_ubi_8.5.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+export MAVEN_OPTS="-Xmx1G"
+alias antlr4='java -Xmx500M -cp "/usr/local/lib/antlr-4.11.1-complete.jar:$CLASSPATH" org.antlr.v4.Tool'
+alias grun='java -Xmx500M -cp "/usr/local/lib/antlr-4.11.1-complete.jar:$CLASSPATH" org.antlr.v4.gui.TestRig'
+
+cd antlr4/
+if ! mvn clean; then
+	if ! mvn -DskipTests install; then
+		echo "Build fails"
+		exit 2
```
Comment (kandarpamalipeddi) [inline a/antlr4/antlr4_v4.11.1_ubi_8.5.sh:54]: If Build fails, `return 1`/`exit 1`.

**Review by kandarpamalipeddi — CHANGES_REQUESTED** [review]

**Comment by muskaan2199** [comment]
@kandarpamalipeddi @nikhil-kalbande made the required changes. Please review. Regarding the user part it's the same in the existing dockerfile hence have done the same made the remaining changes

File: a/antlr4/antlr4_v4.11.1_ubi_8.5.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# Disclaimer: This script has been tested in root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+PACKAGE_NAME=antlr4
+PACKAGE_VERSION=v4.11.1
```
Comment (nikhil-kalbande) [inline a/antlr4/antlr4_v4.11.1_ubi_8.5.sh:22]: @muskaan2199  - can you please make change to accept PACKAGE_VERSION as param.
like PACKAGE_VERSION=${1:-v4.11.1}

**Pull request merged**